### ISSUE TITLE
[Ubuntu] Ruby BUILD FAILED (Because OpenSSL v3 on the New LTS)

### ISSUE THREAD

**Issue by cthonney** [lewagon/setup#383]
Hi guys,
I tested the wagon setup with Ubuntu 22.04(LTS).
Apparently the new version of libssl-dev (**OpenSSL v3 instead of v1.1**) is problematic.
Cannot install Ruby **3.0.3, 2.7.6.** Only Ruby **3.1.2** works
I think the problem will also occur for the WSL setup, because the new Ubuntu LTS is 22.04
This could be a problem for setups.

```
Installing ruby-3.0.3...

BUILD FAILED (Ubuntu 22.04 using ruby-build 20220415)

Inspect or clean up the working tree at /tmp/ruby-build.20220423181649.36535.yjNmS5
Results logged to /tmp/ruby-build.20220423181649.36535.log

Last 10 log lines:
installing bundled gem cache:       /home/tcm/.rbenv/versions/3.0.3/lib/ruby/gems/3.0.0/cache
The Ruby openssl extension was not compiled.
ERROR: Ruby install aborted due to missing extensions
Try running `apt-get install -y libssl-dev` to fetch missing dependencies.

Configure options used:
  --prefix=/home/tcm/.rbenv/versions/3.0.3
  --enable-shared
  LDFLAGS=-L/home/tcm/.rbenv/versions/3.0.3/lib 
  CPPFLAGS=-I/home/tcm/.rbenv/versions/3.0.3/include 
```

`sudo apt-get install -y libssl-dev`
return
`libssl-dev is already the newest version (3.0.2-0ubuntu1).`

**Comment by cthonney** [comment]
A working solution, I compiled OpenSSL 1.1.1
`cd ~/`
`mkdir src`
`cd src`
`wget https://www.openssl.org/source/openssl-1.1.1n.tar.gz`
`tar xvzf openssl-1.1.1n.tar.gz`
`cd openssl-1.1.1n`
`mkdir -p ~/.rubies/openssl-1.1.1n`
`./Configure --prefix=$HOME/.rubies/openssl-1.1.1n --shared`
`./config --prefix=$HOME/.rubies/openssl-1.1.1n --shared `
`make`
`sudo make install`
`RUBY_CONFIGURE_OPTS=--with-openssl-dir=$HOME/.rubies/openssl-1.1.1n rbenv install 3.0.3`

**Comment by dmilon** [comment]
Hi @cthonney :wave: Thank you for pointing this out :pray: This is going to be fixed with the update to Ruby `3.1.2` in https://github.com/lewagon/setup/pull/391
